unblob/processing.py

```diff
@@ -287,6 +289,137 @@ def _process_task(self, result: TaskResult, task: Task):
         _FileTask(self._config, task, stat_report.size, result).process()
 
 
+class DirectoryProcessingError(Exception):
+    ...
+
+
+class _DirectoryTask:
+    def __init__(self, config: ExtractionConfig, dir_task: Task, result: TaskResult):
+        self.config = config
+        self.dir_task = dir_task
+        self.result = result
+
+    def process(self):
+        if self.dir_task.multi_file:
+            # The file has been processed as part of a previous MultiFile
+            return
+
+        logger.debug("Processing directory", path=self.dir_task.path)
+
+        try:
+            processed_paths, extract_dirs = self._process_directory()
+        except DirectoryProcessingError:
+            return
```

Comment:
It is not an exception if everything is handled at the error location. If we do it like this, `DirectoryProcessingError` exceptions are just used for control flow.

We should store the reason (`ErrorReport`) inside the exception at the error location and handle the error here by storing the report and possibly logging it.

```suggestion
        try:
            processed_paths, extract_dirs = self._process_directory()
        except DirectoryProcessingError as e:
            logger.error(e.report, directory=e.directory)
            self.results.add_report(e.report)
            return
```